### PR TITLE
buyback and burn mex tokens instead of burn LP

### PR DESCRIPTION
**Pull request by catalinnnn**

### CODE REVIEW

**Review by mateigraura — APPROVED** [review]

File: dex/elrond_dex_pair/src/lib.rs [hunk 1]
```diff
@@ -317,6 +317,56 @@ pub trait Pair:
...
+        require!(
+            token_in == self.lp_token_identifier().get(),
+            "Wrong liquidity token"
+        );
+
+        let first_token_id = self.first_token_id().get();
+        let second_token_id = self.second_token_id().get();
+
+        let first_token_min_amount = 1u64.into();
+        let second_token_min_amount = 1u64.into();
```
Comment (claudiulataretu) [inline dex/elrond_dex_pair/src/lib.rs:341]: You can use the get equivalent for liquidity view function from pair to determine these values.

File: dex/elrond_dex_pair/src/lib.rs [hunk 2]
```diff
@@ -317,6 +317,56 @@ pub trait Pair:
             .into())
     }
 
+    #[payable("*")]
+    #[endpoint(removeLiquidityAndBuyBackAndBurnToken)]
+    fn remove_liquidity_and_burn_token(
```
Comment (claudiulataretu) [inline dex/elrond_dex_pair/src/lib.rs:322]: Would be an option to enforce that only the farms can call this endpoint? Or a whitelisted address? I'm thinking what would happen if someone, by any chance, would call this method

File: dex/elrond_dex_pair/src/lib.rs (hunk 1 above)
Comment (catalinnnn) [reply dex/elrond_dex_pair/src/lib.rs:341]: What benefit would that bring?

File: dex/elrond_dex_pair/src/lib.rs (hunk 2 above)
Comment (catalinnnn) [reply dex/elrond_dex_pair/src/lib.rs:322]: Not a fan of making the setup even harder. The function has a pretty intuitive name

**Review by claudiulataretu — APPROVED** [review]

**Pull request merged**